src/main/java/dev/jpcode/kits/Kit.java

```diff
@@ -16,16 +19,18 @@ public class Kit {
     private final KitInventory inventory;
     private final long cooldown;
     private @Nullable Item displayItem;
+    private @Nullable LinkedList<String> commands;
```

Comment:
nit: recommend optimizing for read (rather than write) performance, as I generally assume that the number of times a kit is claimed will be >> the number of times the kit is initialized/configured.

```suggestion
    private @Nullable ArrayList<String> commands;
```

Reply:
Quick example of where this comes into play: on `KitCommandsManagerCommand.java:35`

```java
            for (int i = 1; i <= commands.size(); i++) {
                String command = commands.get(i - 1);
```

Since you're currently using a `LinkedList` this loop is actually O(n^2), because indexed lookups (`commands.get(idx)`) are `O(n)` for linked lists. ([the chart of gospel](https://www.bigocheatsheet.com/))

Reply:
Thanks for the advice, was just using a List which doesn't have `.remove(o)` when you go from the Array/Stream to a list using `.asList`.
Replaced this with the first thing I encountered on the internet that did work